### PR TITLE
Raise custom 'MoiraException' on any moira-related error

### LINKED ISSUE #307: Raise exceptions when moira fails
We're getting a fair number of `OSError: write error` exceptions in Sentry. Many, but not all of these seem to be caused by Moira failures and/or timeouts. 

We should add specific error handling for Moira errors so we can track how common they are and determine if we need to handle them differently. For now I'd be satisfied to catch these Moira errors and log them with a unique message so they don't get lumped with other `OSError: write error` messages. 

https://sentry.io/mit-office-of-digital-learning/odl-video-service/issues/349208356/

```
OSError: write error

OSError: write error
```

### PR DESCRIPTION
**Pull request by mbertrand**
#### What are the relevant tickets?
Closes #307

#### What's this PR do?
Raises a `MoiraException` if any error occurs when making calls to the Moira API

#### How should this be manually tested?
- Disconnect wifi
- Log in to OVS using a non-superuser account
- Go to any video detail URL
- You should get a 'MoiraException' the page:
```
MoiraException at /videos/b6491433151144e080e4769c823bdd88/
('Something went wrong with moira client for user %s: %s', 'mattbert@mit.edu', 
'(\'Something went wrong with moira client: %s\', 
"HTTPSConnectionPool(host=\'moiraws.mit.edu\', port=443): Max retries exceeded with url: /moiraws/services/moira?wsdl 
(Caused by NewConnectionError(\'<urllib3.connection.VerifiedHTTPSConnection object at 0x7f78205aca20>: 
Failed to establish a new connection: [Errno -2] Name or service not known\',))")')
```

### CODE REVIEW

File: ui/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+""" Custom exceptions"""
+
+
+class MoiraException(Exception):
+    """Custom exception to be used when something goes wrong with Moira API calls"""
+    pass
```
Comment (noisecapella) [inline ui/exceptions.py]: Can you remove the `pass` here? It's not necessary since the docstring does the same thing here

File: ui/models.py [hunk 2]
```diff
@@ -35,7 +36,10 @@ def members(self):
 
         """
         moira = utils.get_moira_client()
-        return set(moira.list_members(self.name, type=''))
+        try:
+            return set(moira.list_members(self.name, type=''))
+        except Exception as exc:
+            raise MoiraException('Something went wrong with getting moira-list members: %s', str(exc))
```
Comment (noisecapella) [inline ui/models.py]: I don't think the `%s` syntax works here, and it's not necessary to attach the two exceptions. Can you do this instead:

    except Exception as exc:
        raise MoiraException('Something went wrong with getting moira-list members') from exc

File: ui/utils.py [hunk 3]
```diff
@@ -31,7 +33,10 @@ def get_moira_client():
             )
     if errors:
         raise RuntimeError('\n'.join(errors))
-    return Moira(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_PRIVATE_KEY_FILE)
+    try:
+        return Moira(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_PRIVATE_KEY_FILE)
+    except Exception as exc:  # pylint: disable=broad-except
+        raise MoiraException('Something went wrong with creating a moira client: %s', str(exc))
```
Comment (noisecapella) [inline ui/utils.py]: See the exception chaining comment above

File: ui/utils.py [hunk 4]
```diff
@@ -62,12 +67,11 @@ def user_moira_lists(user):
...
     """
     moira_user = get_moira_user(user)
+    moira = get_moira_client()
     try:
-        moira = get_moira_client()
         return moira.user_lists(moira_user.username, moira_user.type)
     except Exception as exc:  # pylint: disable=broad-except
-        log.exception('Something went wrong with the moira client for user %s: %s', user.email, str(exc))
-        return []
+        raise MoiraException('Something went wrong with getting moira lists for user %s: %s', user.email, str(exc))
```
Comment (noisecapella) [inline ui/utils.py]: See exception chaining comment above. To put the user email in the exception message you'll need to use `.format()` or use a format string

File: ui/utils_test.py [hunk 5]
```diff
@@ -64,7 +63,7 @@ def test_user_moira_lists_error(mock_moira_client):
     """
     Test that an empty list is returned if the Moira client can't connect
```
Comment (noisecapella) [inline ui/utils_test.py]: Can you update this text?

File: ui/exceptions.py (hunk 1 above)
Comment (mbertrand) [reply ui/exceptions.py]: OK

File: ui/models.py (hunk 2 above)
Comment (mbertrand) [reply ui/models.py]: OK

File: ui/utils.py (hunk 3 above)
Comment (mbertrand) [reply ui/utils.py]: OK

File: ui/utils.py (hunk 4 above)
Comment (mbertrand) [reply ui/utils.py]: OK

File: ui/utils_test.py (hunk 5 above)
Comment (mbertrand) [reply ui/utils_test.py]: OK

**Pull request merged**